### PR TITLE
luacheck with builtin minetest

### PR DESCRIPTION
**Pull request by BuckarooBanzay**
Proof of Concept for https://github.com/lunarmodules/luacheck/pull/108

Uses the [docker image](https://hub.docker.com/r/mineunit/luacheck) from @S-S-X (thanks)

### CODE REVIEW

**Comment by S-S-X** [comment]
Reminder: for docker image / github action should probably at some point add separate action repo for possible docker builds and reusable actions that define workflow stuff. (didn't plan to actually maintain it long term, just provided for testing / demo etc.)

File: .github/workflows/luacheck.yml [hunk 1]
```diff
@@ -7,4 +7,5 @@ jobs:
       - name: Checkout
         uses: actions/checkout@master
       - name: Luacheck
-        uses: lunarmodules/luacheck@master
\ No newline at end of file
+        run: |
+          docker run --rm -i -u luacheck --mount="type=bind,source=$(pwd),target=/repo" mineunit/luacheck
```
Comment (S-S-X) [inline .github/workflows/luacheck.yml:11]:
```suggestion
        uses: BuckarooBanzay/luacheck@minetest-builtin
```

Should work as there's action and builder defined in repo.
My docker image was mostly meant for easy local tests.

Comment (S-S-X) [reply .github/workflows/luacheck.yml:11]: Actually probably wont work... only checked now and that action is pointing to prebuilt image.
Well, there's still another which uses docker builder:
```suggestion
        uses: S-S-X/minecheck@master
```

**Comment by BuckarooBanzay** [comment]
lunarmodule's luacheck workflow now supports the new standard, i updated the relevant parts, this PR is now ready for review

**Comment by BuckarooBanzay** [comment]
rebased this with current master, should be ready to review/merge now

**Review by S-S-X — APPROVED** [review]
LGTM

**Pull request merged**